### PR TITLE
feat: add plan command

### PR DESCRIPTION
**Pull request by swaingotnochill**
This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (HarikrishnanBalagopal) [inline .gitignore]: Probably we can use this https://github.com/github/gitignore/blob/main/Node.gitignore

File: out/extension.js.map [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"extension.js","sourceRoot":"","sources":["../src/extension.ts"],"names":[],"mappings":";;;AAAA,iCAAiC;AACjC,8DAA2D;AAC3D,yDAAuF;AAEvF,SAAgB,QAAQ,CAAC,OAAgC;IACvD,yBAAyB;IACzB,MAAM,aAAa,GAAG,MAAM,CAAC,MAAM,CAAC,mBAAmB,CAAC,WAAW,CAAC,CAAC;IACrE,IAAA,gDAAkC,EAAC,aAAa,CAAC,CAAC;IAElD,sCAAsC;IACtC,MAAM,qBAAqB,GAAG,MAAM,CAAC,QAAQ,CAAC,eAAe,CAAC,oBAAoB,EAAE,GAAG,EAAE;QACvF,iCAAe,CAAC,MAAM,CAAC,OAAO,CAAC,YAAY,CAAC,CAAC;IAC/C,CAAC,CAAC,CAAC;IAEH,kBAAkB;IAClB,MAAM,WAAW,GAAG,MAAM,CAAC,QAAQ,CAAC,eAAe,CAAC,UAAU,EAAE,sBAAQ,CAAC,CAAC;IAE1E,uBAAuB;IACvB,MAAM,gBAAgB,GAAG,MAAM,CAAC,QAAQ,CAAC,eAAe,CAAC,eAAe,EAAE,GAAG,EAAE;QAC7E,MAAM,CAAC,MAAM,CAAC,sBAAsB,CAAC,8BAA8B,CAAC,CAAC;IACvE,CAAC,CAAC,CAAC;IAEH,uCAAuC;IACvC,OAAO,CAAC,aAAa,CAAC,IAAI,CAAC,qBAAqB,CAAC,CAAC;IAClD,OAAO,CAAC,aAAa,CAAC,IAAI,CAAC,gBAAgB,CAAC,CAAC;IAC7C,OAAO,CAAC,aAAa,CAAC,IAAI,CAAC,WAAW,CAAC,CAAC;AAC1C,CAAC;AAtBD,4BAsBC"}
```
Comment (HarikrishnanBalagopal) [inline out/extension.js.map]: We should .gitignore the files in the build directory called `out/`

File: webview-ui/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import App from "./App";
+
+ReactDOM.render(
```
Comment (HarikrishnanBalagopal) [inline webview-ui/src/index.tsx]: We can use the newer React functions like `createRoot` https://reactjs.org/docs/rendering-elements.html#rendering-an-element-into-the-dom

File: webview-ui/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "hello-world",
```
Comment (HarikrishnanBalagopal) [inline webview-ui/package.json]: change the name to `move2kube-vscode-plugin`

File: webview-ui/package.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "start": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@vscode/webview-ui-toolkit": "^1.2.1",
+    "react": "^17.0.2",
```
Comment (HarikrishnanBalagopal) [inline webview-ui/package.json]: Use the latest react 18

File: webview-ui/package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    "react-dom": "^17.0.2"
```
Comment (HarikrishnanBalagopal) [inline webview-ui/package.json]: use react-dom 18

File: webview-ui/package.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "devDependencies": {
+    "@types/react": "^17.0.33",
```
Comment (HarikrishnanBalagopal) [inline webview-ui/package.json]: same as above

File: webview-ui/package.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    "@types/react-dom": "^17.0.10",
```
Comment (HarikrishnanBalagopal) [inline webview-ui/package.json]: same as above

File: out/extension.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.activate = void 0;
+const vscode = require("vscode");
+const HelloWorldPanel_1 = require("./panels/HelloWorldPanel");
+const getHandlers_1 = require("./utilities/getHandlers");
+function activate(context) {
+    // Create output channel.
```
Comment (HarikrishnanBalagopal) [inline out/extension.js]: The build output directory `out/` needs to be added to `.gitignore`

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
Looks good, have left some comments. PTAL.

**Comment by HarikrishnanBalagopal** [comment]
> This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.

I thought we were focussing on the `transform` command?

**Comment by swaingotnochill** [comment]
> > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> 
> I thought we were focussing on the `transform` command?

Since transform command needs plan, I thought this should be the first step towards it.

**Comment by HarikrishnanBalagopal** [comment]
> > > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> > 
> > 
> > I thought we were focussing on the `transform` command?
> 
> Since transform command needs plan, I thought this should be the first step towards it.

No, we can use the single step flow which does not require plan phase.
```
$ move2kube transform -s ./path/to/src/folder
```

File: package.json [hunk 10]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "mov2kube-vscode-plugin",
```
Comment (HarikrishnanBalagopal) [inline package.json]: typo

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
Just a typo, otherwise looks good. Thanks for making the changes!

File: package.json (hunk 10 above)
Comment (swaingotnochill) [reply package.json]: Okay. I missed it. Sorry for this.

**Comment by swaingotnochill** [comment]
> > > > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> > > 
> > > 
> > > I thought we were focussing on the `transform` command?
> > 
> > 
> > Since transform command needs plan, I thought this should be the first step towards it.
> 
> No, we can use the single step flow which does not require plan phase.
> 
> ```
> $ move2kube transform -s ./path/to/src/folder
> ```

Ohh. I will start to work on it.

**Comment by HarikrishnanBalagopal** [comment]
> > > > > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> > > > 
> > > > 
> > > > I thought we were focussing on the `transform` command?
> > > 
> > > 
> > > Since transform command needs plan, I thought this should be the first step towards it.
> > 
> > 
> > No, we can use the single step flow which does not require plan phase.
> > ```
> > $ move2kube transform -s ./path/to/src/folder
> > ```
> 
> Ohh. I will start to work on it.

Thanks, also as a side note, don't force push to the PR branch. It makes it very hard to see the update without reviewing the entire code again. Github already allows us to squash all the commits when the PR gets accepted and merged.

Also look at conventional commits for a standard way to write commit messages https://www.conventionalcommits.org/en/v1.0.0/#examples

**Review by HarikrishnanBalagopal — APPROVED** [review]
LGTM

**Pull request merged**